osquery/tables/system/linux/memory_info.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <string>
+
+#include <boost/algorithm/string.hpp>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+#include <osquery/filesystem.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+const std::string memInfoPath = {"/proc/meminfo"};
+
+QueryData getMemoryInfo(QueryContext& context) {
+  QueryData results;
+  Row r;
+  r["memory_total"] = INTEGER(-1);
```

Comment:
For values that aren't present, we should bias towards reporting an SQLite `NULL`. Right now that's just "not filling in data" ;)